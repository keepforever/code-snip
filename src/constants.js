export const links = [
  {
    name: "logo-home",
    route: "/",
    display: "Your Snips"
  },
  {
    name: "add-snip-page",
    route: "/add-snip",
    display: "Add Snip"
  },
  {
    name: "data-viz",
    route: "/data-viz",
    display: "Data Viz"
  },
  {
    name: "search-snip",
    route: "/search-snips",
    display: "Search Snips"
  }
];

export const langValueMap = ["","javascript", "css", "python", "html", "graphql"];

export const framValueMap = ["","react", "react-native", "tensorflow"];

export const typeValueMap = ["","boiler", "algo", "visual", "config"]

export const tipLibrary = {
  loginButton: " 1. Send login mutation to GraphQL server 2. Server processes login, fetches user data from Prisma database. 3. Responds with auth token, user info, (snips, name, id, ), then saves to redux store.",
  expansionPanelSummary: "Click here for more details",
  copyToClipboardButton: "Snarf snip to clipboard",
  addSnippitHelp: "Ok, here's the gist...",
  searchBarGuidance: "Scans every line of text you entered into or about a code snippit then returns exact or partial matches.  i.e. search = ' worl ' would match ' world ', but not ' wor ' ",
  homeHelpText: [
    `On the "Your Snips" page your snip records are fetched from the database.`,
    `Starting in the top left and going clockwise, you have snippit “name”, “framework”(if any), “language”(if any), and some randomly extracted, periodically changing “word soup”.`,
    `The “word soup” is intended to help jog your memory as to what the snip might contains.`,
    `For example, if you’re trying to remember how to use toLowerCase() and you can’t remember if it’s ‘arg.toLowerCase()’ or ‘toLowerCase(arg)’, the soup might tell you at a glance if the snippit is a good place to look.`,
    `Press a list item to expand it’s contents.  Click copy to snarf the snip to your clipboard.`,
  ]
}

export const helpSnippitDemo = {
  code: "the code you entered",
  companion:["dependencies or companion libs"],
  framework:"framework",
  id: "cjkglhyx65iww0b78a501j2oa",
  keywords:["some", "keywords"],
  language:"language",
  name:"Name",
  notes:"Notes about snippit",
  reference:["link to examples"],
  bagOfWords: ['the', 'code', 'you', 'entered', 'dependencies', 'companion', 'libs', 'framework', 'some', 'keywords', 'language','Name', 'notes', 'about', 'snippit', 'link', 'to', 'examples'],
}
