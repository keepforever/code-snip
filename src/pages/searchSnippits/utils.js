export const trimSearch = (snipSoup, search) => {
  let match = []
  snipSoup.forEach((item, index) => {
    const bagToString = item.bagOfWords.join(" ")
    if (bagToString.includes(search.trim().toLowerCase())) {
      match.push(index)
    }
  })
  return match
}
