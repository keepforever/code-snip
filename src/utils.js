//search helper
export const processSnipsForSearch = (snips) => {
  let processedSnips = []
  snips.forEach(s => {

    let words = []
    const name = [s.name]
    const notes = s.notes.split(" ").map((w, index) => {
      return w.replace(/[.,\/#!$%\^&\*;:{}=\-_`~()]/g,"");
    })
    const keywords = s.keywords

    const soup = words.concat(name).concat(notes).concat(keywords)
    const id = s.id

    processedSnips.push({
      id,
      bagOfWords: [...soup]
    })
  })

  return processedSnips
}

//dummy data
export const codeString = [
  {
    meta: {
      language: 'javascript',
      uses: 'blah'
    },
    code: `
counterChangeHandler = () => {
  console.log("you did it")
  this.props.onIncrementCounter()
}`
  },
  {
    meta: {
      language: 'javascript',
      uses: 'blah'
    },
    code: `
    .App-header {
      background-color: #222;
      height: 150px;
      padding: 20px;
      color: white;
    }`
  },
];
//helper function to display console logs with more clairity
export const clearLog = (title, varToLog) => {
  console.log(`
    #########################################################
                    ${title}
    #########################################################

  `, varToLog, `


    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    #########################################################

  `, `

  `)
}

// helper functions to distribute state immutably and clean up reducers.
export const updateObject = (oldObject, updatedProperties) => {
  const newObject = {
    ...oldObject,
    ...updatedProperties
  }
  //clearLog('newObject', newObject)
  return newObject
};

// helper function to check validity of form input
export const checkValidity = (value, rules) => {
    let isValid = true;
    if (!rules) {
        return true;
    }

    if (rules.required) {
        isValid = value.trim() !== '' && isValid;
    }

    if (rules.minLength) {
        isValid = value.length >= rules.minLength && isValid
    }

    if (rules.maxLength) {
        isValid = value.length <= rules.maxLength && isValid
    }

    if (rules.isEmail) {
        const pattern = /[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?/;
        isValid = pattern.test(value) && isValid
    }

    if (rules.isNumeric) {
        const pattern = /^\d+$/;
        isValid = pattern.test(value) && isValid
    }

    return isValid;
}
