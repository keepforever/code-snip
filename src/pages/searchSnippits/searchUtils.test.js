import { trimSearch } from './utils'
import { soupTestString } from '../../constants'

it('should return the index of matching inputs', () => {
  const search = "kang"
  const snipSoup = JSON.parse(soupTestString)
  expect(trimSearch(snipSoup, search)[0]).toEqual(13)
})

import SearchSnipsView from './SearchSnipsView';

it('shallow renders correctly', () => {
  // step 1: instanciate component we would like to test
  const wrapper = shallow(
    // this component needs props to render correctly, we need
    // to pass in values for those props here
    <SearchSnipsView
      toggleHelp={true}
      search='kang'
      textChange={() => {} }
      match={[]}
      snipSoup={[]}
      snips={[]}
    />
  );
  // the first time you do a snapshot test a '__snapshots__' folder is
  // created. If you changed something in SearchSnipsView, the test would
  // fail.  If you intended the change, enter 'u' into the test commands
  // and this will update snapshots.
  expect(wrapper).toMatchSnapshot()
});


it('render renders correctly', () => {
  // using 'render()' creates a more granular html output which will
  // allow you inspect the output of, for example, the text within the
  // first div tag that was rendered.
  const wrapper = render(
    <SearchSnipsView
      toggleHelp={true}
      search='kang'
      textChange={() => {} }
      match={[]}
      snipSoup={[]}
      snips={[]}
    />
  );
  // this will add a second output object to the snapshot file
  // more granular than the one created by shallow()
  expect(wrapper).toMatchSnapshot()
});

it('formats some text correctly', () => {
  // the mount() function gives us access to the full DOM and gives access
  // to text properties within tag elements.
  const wrapper = mount(
    <SearchSnipsView
      toggleHelp={true}
      search='kang'
      textChange={() => {} }
      match={[]}
      snipSoup={[]}
      snips={[]}
    />
  );
  // I can find() by the string "CursorDiv" here because in the when I
  // created the CursorDiv styled component, I added a property to it,
  // CursorDiv.displayName = "CursorDiv"
  // the .text() will then return the text within that named object.
  const text = wrapper.find("CursorDiv").text()

  console.log('\n', 'const text from test ', '\n', '\n', text )
  // an example of using enzyme to 'step through the DOM tree' and grab
  // specific things.
  expect(text).toEqual('help')
})

it('calles toggleHelp function correctly', () => {
  // spy() is like a fake function, but it allows you to ask it questions
  // after the fact.  How many times called?  What args were you sent?
  const spy = sinon.spy()

  const wrapper = mount(
    <SearchSnipsView
      toggleHelp={spy}
      search='kang'
      textChange={() => {} }
      match={[]}
      snipSoup={[]}
      snips={[]}
    />
  );
  // simulate click event on CursorDiv
  wrapper.find("CursorDiv").simulate("click")
  // after simulating click event, we can debreif spy()

  expect(spy.calledOnce).toBe(true)
  expect(spy.calledTwice).toBe(false)
})
