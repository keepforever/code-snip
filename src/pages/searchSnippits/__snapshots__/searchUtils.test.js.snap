// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders correctly 1`] = `
<div
  class="sc-dxgOiQ lkmDLs"
>
  <header
    class="MuiPaper-root-9 MuiPaper-elevation4-15 MuiAppBar-root-1 MuiAppBar-positionStatic-5 MuiAppBar-colorDefault-6"
  >
    <div
      class="sc-bdVaJa hFMMqI"
    >
      <h2
        class="MuiTypography-root-36 MuiTypography-title-42 MuiTypography-colorSecondary-57"
      >
        Search Snips
      </h2>
      <aside
        class="MuiTypography-root-36 MuiTypography-body2-44 MuiTypography-colorSecondary-57"
      >
        <div
          class="sc-bZQynM IAXmQ"
        >
          help
        </div>
      </aside>
    </div>
  </header>
  <div
    class="sc-kGXeez kTcdBW"
  >
    <header
      class="MuiPaper-root-9 MuiPaper-elevation4-15 MuiAppBar-root-1 MuiAppBar-positionStatic-5 MuiAppBar-colorDefault-6"
    >
      <div
        class="MuiFormControl-root-61 MuiFormControl-marginNormal-62 MuiFormControl-fullWidth-64"
      >
        <label
          class="MuiFormLabel-root-70 MuiInputLabel-root-65 MuiInputLabel-formControl-66 MuiInputLabel-animated-69 MuiInputLabel-shrink-68"
          data-shrink="true"
          for="search-input"
        >
          Snarf for snips...
        </label>
        <div
          class="MuiInput-root-75 MuiInput-fullWidth-82 MuiInput-formControl-76 MuiInput-underline-79"
        >
          <input
            aria-invalid="false"
            autofocus=""
            class="MuiInput-input-83"
            id="search-input"
            type="text"
            value="kang"
          >
            
          </input>
        </div>
      </div>
    </header>
  </div>
  <div>
    
  </div>
</div>
`;

exports[`shallow renders correctly 1`] = `
<styled.div>
  <WithStyles(AppBar)
    color="default"
    position="static"
  >
    <styled.div>
      <WithStyles(Typography)
        color="secondary"
        variant="title"
      >
        Search Snips
      </WithStyles(Typography)>
      <WithStyles(Typography)
        color="secondary"
        variant="body2"
      >
        <CursorDiv
          onClick={[Function]}
        >
          help
        </CursorDiv>
      </WithStyles(Typography)>
    </styled.div>
  </WithStyles(AppBar)>
  <styled.div>
    <WithStyles(AppBar)
      color="default"
      position="static"
    >
      <TextField
        autoFocus={true}
        fullWidth={true}
        id="search-input"
        label="Snarf for snips..."
        margin="normal"
        onChange={[Function]}
        required={false}
        select={false}
        value="kang"
      />
    </WithStyles(AppBar)>
  </styled.div>
  <div />
</styled.div>
`;
